src/Transport/AzureServiceBusTransport.cs

```diff
@@ -17,20 +17,32 @@ public class AzureServiceBusTransport : TransportDefinition
         /// <summary>
         /// Creates a new instance of <see cref="AzureServiceBusTransport"/>.
         /// </summary>
-        public AzureServiceBusTransport(string connectionString) : base(
+        public AzureServiceBusTransport(string connectionString) : this()
+        {
+            Guard.AgainstNullAndEmpty(nameof(connectionString), connectionString);
+            ConnectionString = connectionString;
+        }
+
+        // Used by the shim API to enable easier migration from the existing API
+        internal AzureServiceBusTransport() : base(
             defaultTransactionMode: TransportTransactionMode.SendsAtomicWithReceive,
             supportsDelayedDelivery: true,
             supportsPublishSubscribe: true,
             supportsTTBR: true)
         {
-            Guard.AgainstNullAndEmpty(nameof(connectionString), connectionString);
-            ConnectionString = connectionString;
         }
 
         /// <inheritdoc />
         public override async Task<TransportInfrastructure> Initialize(HostSettings hostSettings,
             ReceiveSettings[] receivers, string[] sendingAddresses, CancellationToken cancellationToken)
         {
+            // Check if a connection string has been provided when using the shim configuration API.
```

Comment:
Nitpicking: the class can be made partial and this code can be moved to the `obsoletes` file as a separate method so that all stuff to be removed is in a single place.